geode-core/src/main/java/org/apache/geode/distributed/internal/LocatorLoadSnapshot.java

```diff
@@ -448,6 +489,65 @@ private void updateMap(Map map, ServerLocation location, float load, float loadP
     }
   }
 
+  private void updateMap(Map map, ServerLocation location, String memberId, float load,
+      float loadPerConnection) {
+    Map groupMap = (Map) map.get(null);
+    ServerLocationAndMemberId locationAndMemberId =
+        new ServerLocationAndMemberId(location, memberId);
+    LoadHolder holder =
+        (LoadHolder) groupMap.get(locationAndMemberId);
+
+    if (holder != null) {
+      holder.setLoad(load, loadPerConnection);
+    }
+  }
+
+  /**
+   *
+   * @param groupServers the servers to consider
+   * @param excludedServers servers to exclude
+   * @param count how many you want. a negative number means all of them in order of best to worst
+   * @return a list of best...worst server LoadHolders
+   */
+  private List<LoadHolder> findBestServersUsingMemberId(
+      Map<ServerLocationAndMemberId, LoadHolder> groupServers,
+      Set<ServerLocation> excludedServers, int count) {
```

Comment:
I'm wondering about how this method is handling excludedServers. If the hostname-for-senders of all of the servers is the same, doesn't that mean that as soon as one server is in this excluded server list, no results will be returned?

Reply:
You are right, the excluded servers should have the member id into account

Reply:
Ok, the method is not involved in the receiver failover, I was wrong. So, you are right about the behavior of the method. And in the case of configuring same host and port for all the receivers it doesn't seem to be an issue.

Reply:
I'm digging through the code a bit more. I think this method does get called, but maybe we are retrying at a higher level so it doesn't matter for the gateway code.

Reply:
I think it's possible to call this method, but it does look like the WAN will retry and come back with an empty excluded servers set. So I think you're right, it will be ok for the WAN case.